// @ngx-tc/base/esm2020/lib/directives/color/color.directive.mjs
import { Directive, Input, HostBinding, HostListener } from '@angular/core';
import * as i0 from "@angular/core";
export class ColorDirective {
    get class() {
        return 'custom-color';
    }
    ;
    get getStyle() {
        return this.currentColor;
    }
    onMouseEnter() {
        this.currentColor = this.hoveredColor;
    }
    onMouseLeave() {
        this.currentColor = this.defaultColor;
    }
    ngOnInit() {
        this.setColors();
    }
    ngOnChanges() {
        this.setColors();
    }
    setColors() {
        const COLOR = this.tcColor;
        if (!COLOR)
            return null;
        this.defaultColor = (typeof COLOR === 'string') ? COLOR : COLOR[0];
        this.hoveredColor = (typeof COLOR === 'string') ? COLOR : COLOR[1];
        this.currentColor = this.defaultColor;
    }
}
ColorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: ColorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
ColorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.1.3", type: ColorDirective, selector: "[tcColor]", inputs: { tcColor: "tcColor" }, host: { listeners: { "mouseenter": "onMouseEnter()", "mouseleave": "onMouseLeave()" }, properties: { "class": "this.class", "style.color": "this.getStyle" } }, usesOnChanges: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: ColorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[tcColor]'
                }]
        }], propDecorators: { class: [{
                type: HostBinding,
                args: ['class']
            }], getStyle: [{
                type: HostBinding,
                args: ['style.color']
            }], onMouseEnter: [{
                type: HostListener,
                args: ['mouseenter']
            }], onMouseLeave: [{
                type: HostListener,
                args: ['mouseleave']
            }], tcColor: [{
                type: Input
            }] } });
