// @ngx-tc/base/esm2020/lib/directives/border-color/border-color.directive.mjs
import { Directive, HostBinding, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class BorderColorDirective {
    get class() {
        return 'custom-border-color';
    }
    ;
    ngOnInit() {
        const COLOR = this.tcBorderColor;
        this.defaultColor = (typeof COLOR === 'string') ? COLOR : COLOR[0];
        this.hoveredColor = (typeof COLOR === 'string') ? COLOR : COLOR[1];
        this.currentColor = this.defaultColor;
    }
    get getStyle() {
        return this.currentColor;
    }
    onMouseEnter() {
        this.currentColor = this.hoveredColor;
    }
    onMouseLeave() {
        this.currentColor = this.defaultColor;
    }
}
BorderColorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: BorderColorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
BorderColorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.1.3", type: BorderColorDirective, selector: "[tcBorderColor]", inputs: { tcBorderColor: "tcBorderColor" }, host: { listeners: { "mouseenter": "onMouseEnter()", "mouseleave": "onMouseLeave()" }, properties: { "class": "this.class", "style.borderColor": "this.getStyle" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: BorderColorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[tcBorderColor]'
                }]
        }], propDecorators: { class: [{
                type: HostBinding,
                args: ['class']
            }], tcBorderColor: [{
                type: Input
            }], getStyle: [{
                type: HostBinding,
                args: ['style.borderColor']
            }], onMouseEnter: [{
                type: HostListener,
                args: ['mouseenter']
            }], onMouseLeave: [{
                type: HostListener,
                args: ['mouseleave']
            }] } });
