// @ngx-tc/base/esm2020/lib/base.module.mjs
import { NgModule } from '@angular/core';
import { BaseComponent } from './base.component';
import { ShapeDirective } from './directives/shape/shape.directive';
import { ColorDirective } from './directives/color/color.directive';
import { BgColorDirective } from './directives/bg-color/bg-color.directive';
import { BorderColorDirective } from './directives/border-color/border-color.directive';
import { BorderStyleDirective } from './directives/border-style/border-style.directive';
import { GradientDirective } from './directives/gradient/gradient.directive';
import * as i0 from "@angular/core";
const list = [
    BaseComponent,
    ShapeDirective,
    ColorDirective,
    BgColorDirective,
    BorderColorDirective,
    BorderStyleDirective,
    GradientDirective
];
export class TcBaseModule {
}
TcBaseModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: TcBaseModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TcBaseModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: TcBaseModule, declarations: [BaseComponent,
        ShapeDirective,
        ColorDirective,
        BgColorDirective,
        BorderColorDirective,
        BorderStyleDirective,
        GradientDirective], exports: [BaseComponent,
        ShapeDirective,
        ColorDirective,
        BgColorDirective,
        BorderColorDirective,
        BorderStyleDirective,
        GradientDirective] });
TcBaseModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: TcBaseModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: TcBaseModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ...list
                    ],
                    imports: [],
                    exports: [
                        ...list
                    ]
                }]
        }] });
