// @ngx-tc/base/esm2020/lib/directives/font-size/font-size.directive.mjs
import { Directive, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class FontSizeDirective {
    get class() {
        return 'custom-font-size';
    }
    ;
    get getFontSize() {
        const SIZE = this.tcFontSize;
        return (typeof SIZE === 'number') ? SIZE + 'px' : SIZE;
    }
    ;
}
FontSizeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: FontSizeDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
FontSizeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.1.3", type: FontSizeDirective, selector: "[tcFontSize]", inputs: { tcFontSize: "tcFontSize" }, host: { properties: { "class": "this.class", "style.fontSize": "this.getFontSize" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: FontSizeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[tcFontSize]'
                }]
        }], propDecorators: { class: [{
                type: HostBinding,
                args: ['class']
            }], tcFontSize: [{
                type: Input
            }], getFontSize: [{
                type: HostBinding,
                args: ['style.fontSize']
            }] } });
