// @ngx-tc/base/esm2020/lib/directives/gradient/gradient.directive.mjs
import { Directive, Input, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
export class GradientDirective {
    ngOnInit() {
        this.tcGradient.length >= 2 ? this.setGradient(this.tcGradient[0], this.tcGradient[1]) : null;
    }
    get getGradient() {
        return this.gradient;
    }
    get class() {
        return 'custom-gradient';
    }
    ;
    setGradient(firstColor, secondColor) {
        this.gradient = `linear-gradient(to right, ${firstColor} 0%, ${secondColor} 100%)`;
    }
}
GradientDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: GradientDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
GradientDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.1.3", type: GradientDirective, selector: "[tcGradient]", inputs: { tcGradient: "tcGradient" }, host: { properties: { "style.backgroundImage": "this.getGradient", "class": "this.class" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: GradientDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[tcGradient]'
                }]
        }], propDecorators: { tcGradient: [{
                type: Input
            }], getGradient: [{
                type: HostBinding,
                args: ['style.backgroundImage']
            }], class: [{
                type: HostBinding,
                args: ['class']
            }] } });
