// @ngx-tc/base/esm2020/lib/directives/shape/shape.directive.mjs
import { Directive, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class ShapeDirective {
    get class() {
        return 'custom-shape';
    }
    ;
    get getShape() {
        const SHAPE = this.tcShape;
        return (typeof SHAPE === 'number') ? SHAPE + 'px' : SHAPE;
    }
    ;
}
ShapeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: ShapeDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
ShapeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.1.3", type: ShapeDirective, selector: "[tcShape]", inputs: { tcShape: "tcShape" }, host: { properties: { "class": "this.class", "style.borderRadius": "this.getShape" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: ShapeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[tcShape]'
                }]
        }], propDecorators: { class: [{
                type: HostBinding,
                args: ['class']
            }], tcShape: [{
                type: Input
            }], getShape: [{
                type: HostBinding,
                args: ['style.borderRadius']
            }] } });
