// @ngx-tc/base/esm2020/lib/directives/border-style/border-style.directive.mjs
import { Directive, HostBinding, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class BorderStyleDirective {
    get class() {
        return 'custom-border-style';
    }
    ;
    ngOnInit() {
        const STYLE = this.tcBorderStyle;
        this.defaultStyle = (typeof STYLE === 'string') ? STYLE : STYLE[0];
        this.hoveredStyle = (typeof STYLE === 'string') ? STYLE : STYLE[1];
        this.currentStyle = this.defaultStyle;
    }
    get getStyle() {
        return this.currentStyle;
    }
    onMouseEnter() {
        this.currentStyle = this.hoveredStyle;
    }
    onMouseLeave() {
        this.currentStyle = this.defaultStyle;
    }
}
BorderStyleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: BorderStyleDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
BorderStyleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.1.3", type: BorderStyleDirective, selector: "[tcBorderStyle]", inputs: { tcBorderStyle: "tcBorderStyle" }, host: { listeners: { "mouseenter": "onMouseEnter()", "mouseleave": "onMouseLeave()" }, properties: { "class": "this.class", "style.borderStyle": "this.getStyle" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: BorderStyleDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[tcBorderStyle]'
                }]
        }], propDecorators: { class: [{
                type: HostBinding,
                args: ['class']
            }], tcBorderStyle: [{
                type: Input
            }], getStyle: [{
                type: HostBinding,
                args: ['style.borderStyle']
            }], onMouseEnter: [{
                type: HostListener,
                args: ['mouseenter']
            }], onMouseLeave: [{
                type: HostListener,
                args: ['mouseleave']
            }] } });
